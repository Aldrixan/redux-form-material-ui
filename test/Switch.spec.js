import { render, unmountComponentAtNode } from 'react-dom';

import React from "react"
import ReduxFormMaterialUISwitch from "../src/Switch"
import { act } from 'react-dom/test-utils';
import noop from "lodash.noop"

// import renderer from "react-test-renderer"

describe( "Switch", () => {
    let container = null;
    beforeEach( () => {
        container = document.createElement( 'div' );
        document.body.appendChild( container );
    } )

    afterEach( () => {
        unmountComponentAtNode( container );
        container.remove();
        container = null;
    } );

    test( 'renders an unchecked Switch', () => {
        act( () => {
            render( <ReduxFormMaterialUISwitch input={{ onChange: noop, name: "mySwitch" }} />, container );
        } )
        let element = container.querySelector( 'input[name="mySwitch"]' );
        expect( element ).toBeDefined();
        expect( element.checked ).toBeFalsy()
    } )

    test( 'renders a checked Switch', () => {
        act( () => {
            render( <ReduxFormMaterialUISwitch input={{ value: true, onChange: noop, name: "mySwitch" }} />, container );
        } )
        let element = container.querySelector( 'input[name="mySwitch"]' );
        expect( element.checked ).toBeTruthy()
    } )

    test( 'should ignore checked', () => {
        act( () => {
            render( <ReduxFormMaterialUISwitch input={{ onChange: noop, name: "mySwitch" }} checked={true} />, container );
        } )
        let element = container.querySelector( 'input[name="mySwitch"]' );
        expect( element.checked ).toBeFalsy()
    } )

    test( 'should ignore defaultChecked', () => {
        act( () => {
            render( <ReduxFormMaterialUISwitch input={{ onChange: noop, name: "mySwitch" }} defaultChecked={true} />, container );
        } )
        let element = container.querySelector( 'input[name="mySwitch"]' );
        expect( element.checked ).toBeFalsy();
    } )
} )