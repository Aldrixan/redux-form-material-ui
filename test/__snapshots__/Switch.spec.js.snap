// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders a checked Switch 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiSwitch-root-5"
  >
    <span
      className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorSecondary-11 MuiSwitchBase-checked-15 MuiSwitch-checked-9"
      onBlur={[Function]}
      onCheck={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-23"
      >
        <span
          className="MuiSwitch-icon-6 MuiSwitch-iconChecked-7"
        />
        <input
          checked={true}
          className="MuiSwitchBase-input-17"
          name="mySwitch"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-27"
      />
    </span>
    <span
      className="MuiSwitch-bar-13"
    />
  </span>
  <span
    className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
  />
</label>
`;

exports[`Switch renders an unchecked Switch 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiSwitch-root-5"
  >
    <span
      className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorSecondary-11"
      onBlur={[Function]}
      onCheck={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-23"
      >
        <span
          className="MuiSwitch-icon-6"
        />
        <input
          checked={false}
          className="MuiSwitchBase-input-17"
          name="mySwitch"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-27"
      />
    </span>
    <span
      className="MuiSwitch-bar-13"
    />
  </span>
  <span
    className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
  />
</label>
`;

exports[`Switch should ignore checked 1`] = `
<label
  className="MuiFormControlLabel-root-1"
>
  <span
    className="MuiSwitch-root-5"
  >
    <span
      className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorSecondary-11"
      onBlur={[Function]}
      onCheck={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-23"
      >
        <span
          className="MuiSwitch-icon-6"
        />
        <input
          checked={false}
          className="MuiSwitchBase-input-17"
          name="mySwitch"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-27"
      />
    </span>
    <span
      className="MuiSwitch-bar-13"
    />
  </span>
  <span
    className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
  />
</label>
`;

exports[`Switch should ignore defaultChecked 1`] = `
<label
  className="MuiFormControlLabel-root-1"
  defaultChecked={true}
>
  <span
    className="MuiSwitch-root-5"
  >
    <span
      className="MuiButtonBase-root-24 MuiIconButton-root-18 MuiSwitchBase-root-14 MuiSwitch-switchBase-8 MuiSwitch-colorSecondary-11"
      onBlur={[Function]}
      onCheck={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-23"
      >
        <span
          className="MuiSwitch-icon-6"
        />
        <input
          checked={false}
          className="MuiSwitchBase-input-17"
          name="myCheckbox"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-27"
      />
    </span>
    <span
      className="MuiSwitch-bar-13"
    />
  </span>
  <span
    className="MuiTypography-root-34 MuiTypography-body1-43 MuiFormControlLabel-label-4"
  />
</label>
`;
